jetty-core/jetty-http2/jetty-http2-server/src/main/java/org/eclipse/jetty/http2/server/internal/HttpStreamOverHTTP2.java

```diff
@@ -82,7 +85,12 @@ public Runnable onRequest(HeadersFrame frame)
         {
             _requestMetaData = (MetaData.Request)frame.getMetaData();
 
+            ComplianceViolation.Listener listener = Server.getComplianceViolationListener(_httpChannel.getConnectionMetaData().getConnector()).initialize();
```

Comment:
Don't think this is right. By the time we are here, the channel has been initialized/recycled already, so we already have the listener.